bridges/relays/finality/src/finality_loop.rs

```diff
@@ -302,17 +326,28 @@ impl<P: FinalitySyncPipeline, SC: SourceClient<P>, TC: TargetClient<P>> Finality
 		);
 
 		// read missing headers
-		let selector = JustifiedHeaderSelector::new::<SC, TC>(&self.source_client, info).await?;
+		let selector = JustifiedHeaderSelector::new::<SC, TC>(
+			&self.source_client,
+			info,
+			self.sync_params.headers_to_relay,
+			free_headers_interval,
+		)
+		.await?;
 		// if we see that the header schedules GRANDPA change, we need to submit it
-		if self.sync_params.only_mandatory_headers {
+		if self.sync_params.headers_to_relay == HeadersToRelay::Mandatory {
```

Comment:
What about allowing here also `HeadersToRelay::All`?

Reply:
Sorry, I don't get what you mean. Can you, please, elaborate? `All` and `Free` are handled later, in the `selector.select` call.

Reply:
nvm, I need to walk through the entire PR first :)